pkg/ccl/changefeedccl/changefeed_processors.go

```diff
@@ -1479,8 +1479,11 @@ func (cf *changeFrontier) Start(ctx context.Context) {
 
 	if err := checkpoint.Restore(cf.frontier, cf.spec.SpanLevelCheckpoint); err != nil {
 		if log.V(2) {
-			log.Dev.Infof(cf.Ctx(), "change frontier encountered error on checkpoint restore: %v", err)
+			log.Dev.Infof(cf.Ctx(),
+				"change frontier moving to draining due to error restoring span-level checkpoint: %v", err)
 		}
+		cf.MoveToDraining(err)
```

Comment:
In the issue, you point out that L635 (L664 in this file) only returns an error when restore returns an error. Should that also move to draining?

Reply:
It gets moved to draining higher up in the call stack (just edited the issue to make that more clear) https://github.com/cockroachdb/cockroach/blob/96631ad731561f563f02f7fb791b75458aa421ba/pkg/ccl/changefeedccl/changefeed_processors.go#L343-L351